core/redis_client.py

```
@@ -0,0 +1,441 @@
+            logger.info(f"Redis async pool initialized: {self.host}:{self.port}")
+        except Exception as e:
+            logger.error(f"Failed to initialize Redis pool: {e}")
+            raise
+
+    async def retry(self, coro, retries=3, delay=0.2):
+        """Retry a coroutine with exponential backoff"""
+        last_exc = None
+        for attempt in range(retries):
+            try:
+                return await coro
+            except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)